packages/client/src/comp/states/BaseContainer.ts

```diff
@@ -63,13 +64,24 @@ const useFields = () => {
     mailing: { valid: true, value: '' },
   })
 
-   /**
-   * Detects the content of all inputs needed for checking voter registration.
-   * When possible, this function will contact the server, updating `registrationStatus`
-   * when finished.
-   */
+  const nameParts: NameParts = {
+    first: fields.firstName.value,
+    middle: fields.middleName.value ? fields.middleName.value : undefined,
+    last: fields.lastName.value,
+    suffix: fields.suffix.value ? fields.suffix.value : undefined,
+  }
+
   const updateField = (id: InputId, value: string) => {
     _updateValid({ ...fields, [id]: { valid: isInputValid(id, value), value } })
+    if (id.indexOf('Name') !== -1 || id === 'suffix') {
+      // Makes firstName => first, middleName => middle, etc.
+      const key = id.replace('Name', '') as keyof NameParts
+      if (key === 'middle' || key === 'suffix') {
+        nameParts[key] = value ? value : undefined
```

Comment:
I'm 90% sure this should update the state, not the variable (which doesn't do anything)?

Reply:
It's updating both the state and the variable so when we call `nameParts` at `Base.tsx` we don't have to make it a function to get the latest values. By the way, we don't use `nameParts` directly for the values in this container since it lacks a `valid` boolean indicator.

Reply:
Got it -- that was a bit subtle.  I don't think you need to update the variable.  I think what happens (but check and maybe write a test for it), is that updating the state 'triggers' a rerender so the variable will be recalculated again.

Sorry -- was this why you wanted to make `nameParts` a function?  I still don't think you have to.